packages/tool-cache/src/tool-cache.ts

```diff
@@ -202,9 +202,36 @@ export async function extractTar(
     throw new Error("parameter 'file' is required")
   }
 
+  // Create dest
   dest = await _createExtractFolder(dest)
-  const tarPath: string = await io.which('tar', true)
-  await exec(`"${tarPath}"`, [flags, '-C', dest, '-f', file])
+
+  // Determine whether GNU tar
+  let versionOutput = ''
+  await exec('tar --version', [], {
+    ignoreReturnCode: true,
+    listeners: {
+      stdout: (data: Buffer) => (versionOutput += data.toString()),
+      stderr: (data: Buffer) => (versionOutput += data.toString())
+    }
+  })
+  const isGnuTar = versionOutput.toUpperCase().includes('GNU TAR')
```

Comment:
Consider debugging this so we can troubleshoot easier if for some reason this breaks. (Analyzing text output of a cli can be a little fragile)

Reply:
We may also want to debug/warn if we don't find gnu tar on windows that we would recommend that you setup your self hosted runner with gnu tar on windows in the path.

Reply:
the output flows to the console too, so we're good wrt that aspect